app/services/email_checker.rb

```diff
@@ -0,0 +1,89 @@
+class EmailChecker
+  extend Forwardable
+  def_delegators :SendgridApi, :bounced?, :spam_reported?
+
+  def initialize(original_address, override_sendgrid = false)
+    @original_address = original_address
+    @parsed = parse_address(original_address)
+    @override_sendgrid = override_sendgrid
+  end
+
+  def error
```

Comment:
Oh how I wish there weren't multiple, clashing, and incompatible versions of `delegate`.